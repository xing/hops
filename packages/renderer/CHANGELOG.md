# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="9.7.0"></a>
# [9.7.0](https://github.com/xing/hops/compare/v9.6.1...v9.7.0) (2018-01-22)




**Note:** Version bump only for package hops-renderer

<a name="9.5.0"></a>
# [9.5.0](https://github.com/xing/hops/compare/v9.4.3...v9.5.0) (2018-01-09)




**Note:** Version bump only for package hops-renderer

<a name="9.4.2"></a>
## [9.4.2](https://github.com/xing/hops/compare/v9.4.1...v9.4.2) (2018-01-08)




**Note:** Version bump only for package hops-renderer

<a name="9.4.1"></a>
## [9.4.1](https://github.com/xing/hops/compare/v9.4.0...v9.4.1) (2018-01-08)




**Note:** Version bump only for package hops-renderer

<a name="9.3.0"></a>
# [9.3.0](https://github.com/xing/hops/compare/v9.2.0...v9.3.0) (2017-12-14)




**Note:** Version bump only for package hops-renderer

<a name="9.1.1"></a>
## [9.1.1](https://github.com/xing/hops/compare/v9.1.0...v9.1.1) (2017-12-05)




**Note:** Version bump only for package hops-renderer

<a name="8.0.0"></a>
# [8.0.0](https://github.com/xing/hops/compare/v7.4.1...v8.0.0) (2017-11-22)


### Features

* **renderer:** add support for an options hash ([8206ad1](https://github.com/xing/hops/commit/8206ad1))
* **renderer:** make renderer use bootstrapServer config ([ce2298d](https://github.com/xing/hops/commit/ce2298d))


### BREAKING CHANGES

* **renderer:** createRenderer signature changed to options hash instead of discrete args




<a name="7.2.0"></a>
# [7.2.0](https://github.com/xing/hops/compare/v7.1.0...v7.2.0) (2017-10-18)




**Note:** Version bump only for package hops-renderer

<a name="7.1.0"></a>
# [7.1.0](https://github.com/xing/hops/compare/v7.0.0...v7.1.0) (2017-10-16)




**Note:** Version bump only for package hops-renderer

<a name="7.0.0"></a>
# [7.0.0](https://github.com/xing/hops/compare/v6.2.8...v7.0.0) (2017-10-13)


### Features

* **renderer:** split hops packages, introduce changelog
