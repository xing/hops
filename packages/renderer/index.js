'use strict';

var events = require('events');

var express = require('express');
var mocks = require('node-mocks-http');

var hopsConfig = require('hops-config');
var createMiddleware = require('hops-middleware');

module.exports = function createRenderer(webpackConfig, watchOptions) {
  if (arguments.length === 1 && 'webpackConfig' in webpackConfig) {
    webpackConfig = webpackConfig.webpackConfig;
    watchOptions = webpackConfig.watchOptions;
    console.warn(
      'Calling createRenderer() with just an options hash is deprecated.',
      'Please refer to the latest docs:',
      'https://github.com/xing/hops/blob/master/packages/renderer/README.md'
    );
  }
  var router = new express.Router();
  if (hopsConfig.bootstrapServer) {
    hopsConfig.bootstrapServer(router, hopsConfig);
  }
  router.use(createMiddleware(webpackConfig, watchOptions));
  return function(options) {
    if (typeof options === 'string') {
      options = { url: options };
    }
    return new Promise(function(resolve, reject) {
      var req = mocks.createRequest(options);
      var res = mocks.createResponse({
        eventEmitter: events.EventEmitter,
        request: req,
      });
      res.on('finish', function() {
        if (res.statusCode !== 200) {
          reject(new Error('invalid status code: ' + res.statusCode));
        } else {
          resolve(res._getData());
        }
      });
      router.handle(req, res, function(error) {
        if (error) {
          reject(error);
        } else {
          resolve();
        }
      });
    });
  };
};
