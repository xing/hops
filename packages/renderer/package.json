{
  "name": "hops-renderer",
  "version": "10.0.0-rc.0",
  "description": "Hops HTML renderer based on Webpack",
  "keywords": ["hops", "webpack"],
  "license": "MIT",
  "main": "index.js",
  "files": ["index.js"],
  "dependencies": {
    "hops-config": "10.0.0-rc.0",
    "hops-middleware": "10.0.0-rc.0",
    "node-mocks-http": "^1.6.6"
  }
}
