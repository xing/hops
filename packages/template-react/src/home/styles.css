:root {
  --font-family-sans: 'Helvetica Neue', Arial, sans;
  --font-size: 2rem;
  --primary-color: rebeccapurple;
}

body {
  text-align: center;
  color: var(--primary-color);
}

.headline {
  margin-top: 2em;
  display: inline-block;

  font-family: var(--font-family-sans);
  font-size: var(--font-size);
}

.headline::after {
  content: ' ðŸš€';
}
