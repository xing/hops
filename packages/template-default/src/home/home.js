// @flow
import React from 'react';
import Helmet from 'react-helmet';
import styles from './styles.css';

export default function Home() {
  return (
    <div>
      <Helmet>
        <title>Hops Demo</title>
      </Helmet>
      <h1 className={styles.headline}>
        Hello World!
      </h1>
    </div>
  );
}
