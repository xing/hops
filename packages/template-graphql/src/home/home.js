import React from 'react';
import Helmet from 'react-helmet';
import { graphql } from 'react-apollo';

import styles from './styles.css';
import query from './commits.gql';

const withCommits = graphql(query);

export const Home = ({ data: { loading, github: { repo } = {} } }) => (
  <div>
    <Helmet>
      <title>Hops Demo</title>
    </Helmet>
    <h1 className={styles.headline}>Hello World!</h1>
    {loading ? (
      <div>loading commits...</div>
    ) : (
      repo.commits.map(commit => {
        const repoUrl = `https://github.com/${repo.owner.login}/${repo.name}`;
        return (
          <p key={commit.sha}>
            <a href={`${repoUrl}/commit/${commit.sha}`}>"{commit.message}"</a>{' '}
            by <b>{commit.author.login || commit.author.name}</b>
          </p>
        );
      })
    )}
  </div>
);

export default withCommits(Home);
