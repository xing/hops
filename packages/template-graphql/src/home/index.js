import React from 'react';
import { graphql } from '@apollo/client/react/hoc';
import { Helmet } from 'react-helmet-async';
import query from './jobs.gql';
import styles from './styles.css';

const withJobs = graphql(query);

export const Home = ({ data: { loading, error, jobSearchByQuery } }) => {
  return (
    <div>
      <Helmet>
        <title>Hops Demo</title>
      </Helmet>
      <h1 className={styles.headline}>Hello World!</h1>
      {loading ? (
        <div>loading...</div>
      ) : (
        jobSearchByQuery.collection.map(({ jobDetail }) => {
          const company = jobDetail.companyInfo.company;
          return (
            <p key={jobDetail.id}>
              <a href={jobDetail.url}>"{jobDetail.title}"</a> at{' '}
              <b>{company ? company.companyName : 'N/A'}</b>
            </p>
          );
        })
      )}
    </div>
  );
};

export default withJobs(Home);
