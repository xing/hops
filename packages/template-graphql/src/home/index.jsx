import React from 'react';
import { useQuery } from '@apollo/client';
import { Helmet } from 'react-helmet-async';
import query from './jobs.gql';
import styles from './styles.css';

export const Home = ({ loading, data: { jobSearchByQuery } = {} }) => {
  return (
    <div>
      <Helmet>
        <title>Hops Demo</title>
      </Helmet>
      <h1 className={styles.headline}>Hello World!</h1>
      {loading ? (
        <div>loading...</div>
      ) : (
        jobSearchByQuery.collection.map(({ jobDetail }) => {
          const company = jobDetail.companyInfo.company;
          return (
            <p key={jobDetail.id}>
              <a href={jobDetail.url}>&quot;{jobDetail.title}&quot;</a> at{' '}
              <b>{company ? company.companyName : 'N/A'}</b>
            </p>
          );
        })
      )}
    </div>
  );
};

export default function HomeWithData() {
  const { data, loading } = useQuery(query);

  return <Home data={data} loading={loading} />;
}
