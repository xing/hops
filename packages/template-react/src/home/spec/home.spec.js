// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import { Home } from '../home';

it('renders loading state correctly', () => {
  const tree = renderer.create(
    <Home data={{ loading: true }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders loaded state correctly', () => {
  const data = {
    loading: false,
    github: {
      repo: {
        commits: [{
          sha: 'foo',
          message: 'bar',
          author: {
            login: 'baz'
          }
        }]
      }
    }
  };
  const tree = renderer.create(
    <Home data={ data } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
