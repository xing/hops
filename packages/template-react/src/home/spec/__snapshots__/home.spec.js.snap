// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loaded state correctly 1`] = `
<div>
  <h1
    className="headline"
  >
    Hello World!
  </h1>
  <p>
    "
    bar
    " by 
    <b>
      baz
    </b>
  </p>
</div>
`;

exports[`renders loading state correctly 1`] = `
<div>
  <h1
    className="headline"
  >
    Hello World!
  </h1>
  <p>
    Loading...
  </p>
</div>
`;
